### PR TITLE
Feature/bump cmake version

### PR DESCRIPTION
**Pull request by goedderz**
### Scope & Purpose

Minor cmake changes, preparations for https://github.com/arangodb/arangodb/pull/9596.

cmake 3.9 is needed for IPO.

### CODE REVIEW

**Comment by goedderz** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/5602/

File: CMakeLists.txt [hunk 1]
```diff
@@ -275,6 +263,7 @@ set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 option(STATIC_EXECUTABLES "produce static executables" OFF)
 if (STATIC_EXECUTABLES)
   set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -static")
+  set(CMAKE_FIND_LIBRARY_SUFFIXES ".a")
```
Comment (dothebart) [inline CMakeLists.txt]: this won't work on windows, its .lib there

File: CMakeLists.txt [hunk 2]
```diff
@@ -314,6 +303,7 @@ set(ARANGO_VPACK_FRIENDLY_STRING   "arangovpack - vpack printer")
 set(LIB_ARANGO        arango)
 set(LIB_ARANGO_GEO    arango_geo)
 set(LIB_ARANGO_V8     arango_v8)
+set(LIB_ARANGOSERVER  arangoserver)
```
Comment (dothebart) [inline CMakeLists.txt]: @ObiWahn says: don't do this.

File: arangod/CMakeLists.txt [hunk 3]
```diff
@@ -626,17 +626,17 @@ include(ClusterEngine/CMakeLists.txt)
 include(RocksDBEngine/CMakeLists.txt)
 include(MMFiles/CMakeLists.txt)
 
-add_library(arangoserver STATIC 
+add_library(${LIB_ARANGOSERVER} STATIC
```
Comment (dothebart) [inline arangod/CMakeLists.txt]: @ObiWahn says: don't do this.

**Comment by goedderz** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/5611/

**Comment by goedderz** [comment]
Jenkins is green.

**Pull request merged**